### PR TITLE
These commits update the v8  add ClusterClient support.

### PR DESCRIPTION
**Pull request by vvlgo**
This change is [Reviewable](https://reviewable.io/reviews/elliotchance/redismock/41)

### CODE REVIEW

File: v8/client.go [hunk 1]
```diff
@@ -17,7 +18,61 @@ type ClientMock struct {
...
+
+func (m *ClientMock) Do(ctx context.Context, args ...interface{}) *redis.Cmd {
+	if !m.hasStub("Watch") {
+		return m.client.Do(ctx, args)
+	}
+	return m.Called().Get(0).(*redis.Cmd)
+}
+
+func (m *ClientMock) Process(ctx context.Context, cmd redis.Cmder) error {
+	if !m.hasStub("Watch") {
```
Comment (elliotchance) [inline v8/client.go]: "Process"

File: v8/client.go [hunk 2]
```diff
@@ -17,7 +18,61 @@ type ClientMock struct {
 
 	mock.Mock
 	redis.Cmdable
-	client *redis.Client
+	client redis.UniversalClient
+	ctx    context.Context
```
Comment (elliotchance) [inline v8/client.go]: How does this get set if the test intends to mock `Context()`? Perhaps call it `Ctx`?

File: v8/client.go [hunk 3]
```diff
@@ -17,7 +18,61 @@ type ClientMock struct {
...
+func (m *ClientMock) Context() context.Context {
+	return m.ctx
+}
+
+func (m *ClientMock) AddHook(hook redis.Hook) {
+	m.client.AddHook(hook)
+}
+
+func (m *ClientMock) Do(ctx context.Context, args ...interface{}) *redis.Cmd {
+	if !m.hasStub("Watch") {
```
Comment (elliotchance) [inline v8/client.go]: "Do"

File: v8/client.go [hunk 4]
```diff
@@ -17,7 +18,61 @@ type ClientMock struct {
...
+func (m *ClientMock) PSubscribe(ctx context.Context, channels ...string) *redis.PubSub {
+
+	if !m.hasStub("PSubscribe") {
+		return m.client.PSubscribe(ctx, channels...)
+	}
+	return m.Called().Get(0).(*redis.PubSub)
+}
+
+func (m *ClientMock) Close() error {
+	if !m.hasStub("Watch") {
```
Comment (elliotchance) [inline v8/client.go]: "Close"

**Review by elliotchance — CHANGES_REQUESTED** [review]

**Comment by vvlgo** [comment]
Yes, you are right. I corrected it. Please review this

**Review by elliotchance — APPROVED** [review]

**Pull request merged**